prime-dotnet-webapi/Services/SiteService.cs

```diff
@@ -42,12 +42,30 @@ public async Task<bool> PecAssignableAsync(int siteId, string pec)
                 })
                 .SingleAsync();
 
-            if (siteDto.CareSettingCode == (int)CareSettingType.HealthAuthority
-                || siteDto.PEC == pec)
+            if (siteDto.PEC == pec)
             {
                 return true;
             }
 
+            if (siteDto.CareSettingCode == (int)CareSettingType.HealthAuthority)
+            {
+                var healthAuthorityDto = await _context.HealthAuthoritySites
```

Comment:
Since you are jsut returning a single field you don't need to make a DTO you can just return the HealthAuthorityOrganizationId

Reply:
Changed to only return the `HealthAuthorityOriganizationId`